kubetest2/kubetest2-gce/deployer/up.go

```diff
@@ -135,3 +93,75 @@ func enableComputeAPI(project string) error {
 
 	return nil
 }
+
+// getProjectFromBoskos creates a boskos client, acquires a gcp project
+// and starts a heartbeat goroutine to keep the project reserved
+func (d *deployer) getProjectFromBoskos() error {
+	boskos, err := client.NewClient(
+		os.Getenv("JOB_NAME")+"-kubetest2",
+		d.BoskosLocation,
+		"",
+		"",
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create boskos client: %s", err)
+	}
+
+	resourceType := "gce-project"
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*5)
+	defer cancel()
+
+	boskosProject, err := boskos.AcquireWait(ctx, resourceType, "free", "busy")
+	if err != nil {
+		return fmt.Errorf("failed to get a %q from boskos: %s", resourceType, err)
+	}
+	if boskosProject == nil {
+		return fmt.Errorf("boskos had no %s available", resourceType)
+	}
+
+	startBoskosHeartbeat(
+		boskos,
+		boskosProject,
+		time.Duration(d.BoskosAcquireTimeoutSeconds)*time.Second,
+		d.boskosHeartbeatClose,
+	)
+
+	d.boskos = boskos
+	d.boskosProject = boskosProject
+	d.GCPProject = boskosProject.Name
+
+	return nil
+}
+
+// startBoskosHeartbeat starts a goroutine that sends periodic updates to boskos
+// about the provided resource until the channel is closed. This prevents
+// boskos from taking the resource from the deployer while it is still in use.
+func startBoskosHeartbeat(c *client.Client, resource *boskosCommon.Resource, interval time.Duration, close chan struct{}) {
+	go func(c *client.Client, resource *boskosCommon.Resource) {
+		for {
+			select {
+			case <-close:
+				klog.Info("Boskos heartbeat func received signal to close")
+				return
+			case <-time.Tick(interval):
+				klog.Info("Sending heartbeat to Boskos")
```

Comment:
I think this should be at a lower verbosity level like debug.
https://godoc.org/github.com/kubernetes/klog#V

let's change all info to V(1) and debug to V(2)

Reply:
I get the concept of V, but to set it it looks like it expects flags to be parsed, but kubetest2 uses a different flag parser. Do you know how to get around this?

https://godoc.org/github.com/kubernetes/klog#InitFlags

Reply:
Done, was a bit confused about how to add to the flag set. Tested at all verbosity levels.